app/src/main/java/com/itsaky/androidide/utils/SortUtils.java

```diff
@@ -0,0 +1,173 @@
+package com.itsaky.androidide.utils;
+
+import com.blankj.utilcode.util.FileUtils;
+
+import com.itsaky.androidide.app.StudioApp;
+import com.itsaky.androidide.managers.PreferenceManager;
+import java.io.File;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+public class SortUtils {
+
+  private SortUtils() {}
+
+  private static final int SORT_ALPHA = 0;
+  private static final int SORT_TYPE = 1;
+  private static final int SORT_SIZE = 2;
+  private static final int SORT_DATE = 3;
+
+  public static void sortList(List<String> content, String current) {
+    PreferenceManager manager = StudioApp.getInstance().getPrefManager();
+
+    int len = content != null ? content.size() : 0;
+
+    if (len == 0) return;
+
+    int index = 0;
+    String[] items = new String[len];
+    content.toArray(items);
+
+    switch (manager.getInt("pref_sort_file", 1)) {
+      case SORT_ALPHA:
+        Arrays.sort(items, Comparator_ALPH);
+        content.clear();
+
+        Collections.addAll(content, items);
+        break;
+      case SORT_SIZE:
+        Arrays.sort(items, Comparator_SIZE);
+        content.clear();
+
+        for (String a : items) {
+          if (new File(current + "/" + a).isDirectory()) content.add(index++, a);
+          else content.add(a);
+        }
+        break;
```

Comment:
You can create a simple `Comparator` which does the sorting based on size and type of `File`. See [this](https://github.com/itsaky/AndroidIDE/blob/222fa49ee15d561d385ff9184feb344e022fe1e7/app/src/main/java/com/itsaky/androidide/tasks/callables/FileTreeCallable.java#L52) for example. It shorts the files alphabetically and then with their type.

You could even use `Comparatior.comparing(Function, Comparator).thenComparing(Function, Comparator)` for even faster sorting. Same goes for below two cases.

Edit:
I see that the below comparators (`Comparator_SIZE` and `Comparator_TYPE`) handle the directory ordering as well. Could you explain why you're trying to order directories here?

Can we not simply use `content.sort(Comparator_SIZE)` here?

Reply:
It's not used class. Add by my mistake

Reply:
Okay.